utils/command/command.go

```diff
@@ -44,18 +57,85 @@ func (c *Command) Run(ctx context.Context) (*CommandResult, error) {
 		cmd.Dir = c.WorkDir
 	}
 
+	var stdOut, stdErr bytes.Buffer
 	cmd.Stdout = &stdOut
 	cmd.Stderr = &stdErr
 
-	err = cmd.Run()
+	err := cmd.Start()
 
-	return &CommandResult{
-		cmd.ProcessState,
-		&stdOut,
-		&stdErr,
+	return &State{
+		cmd:    cmd,
+		StdOut: &stdOut,
+		StdErr: &stdErr,
 	}, err
 }
 
-func (c Command) String() string {
+func (c *Command) String() string {
 	return fmt.Sprintf("Command{Cmd: %s, Args: %s, Env: %s, WorkDir: %s}", c.Cmd, c.Args, c.Env, c.WorkDir)
 }
+
+type State struct {
```

Comment:
any reason why the method for command lifecycle control are implemented on top of the `State` object rather than `Command`?
We could have only extended the `Command` itself

Reply:
The reason for having a separate state is to allow running the `Command` multiple times even in parallel.